src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs

```diff
@@ -0,0 +1,212 @@
+﻿using Microsoft.TestPlatform.VsTestConsole.TranslationLayer;
+using Microsoft.TestPlatform.VsTestConsole.TranslationLayer.Interfaces;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Serilog.Events;
+using Stryker.Core.ExecutableFinders;
+using Stryker.Core.Initialisation;
+using Stryker.Core.Options;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace Stryker.Core.TestRunners.VsTest
+{
+    public class VsTestRunner : ITestRunner
+    {
+        public bool Running { get; private set; }
+        private readonly object runLock = new object();
+
+        private readonly StrykerOptions _options;
+        private readonly ProjectInfo _projectInfo;
+
+        private readonly IVsTestConsoleWrapper _vsTestConsole;
+
+        private readonly VsTestHelper _vsTestHelper;
+        private List<string> _messages = new List<string>();
+
+        private IEnumerable<string> _sources;
+        private IEnumerable<TestCase> _testCases;
+
+        public VsTestRunner(StrykerOptions options, ProjectInfo projectInfo, IEnumerable<TestCase> testCases)
+        {
+            _options = options;
+            _projectInfo = projectInfo;
+            _testCases = testCases;
+            _vsTestHelper = new VsTestHelper(options);
+
+            _vsTestConsole = PrepareVsTestConsole();
+
+            InitializeVsTestConsole();
+        }
+
+        public TestRunResult RunAll(int? timeoutMS, int? activeMutationId)
+        {
+            TestRunResult testResult = new TestRunResult() { Success = false };
+            lock (runLock)
+            {
+                Running = true;
+
+                var testResults = RunAllTests(activeMutationId, GenerateRunSettings(timeoutMS ?? 0));
+
+                // For now we need to throw an OperationCanceledException when a testrun has timed out. 
+                // We know the testrun has timed out because we received less test results from the test run than there are test cases in the unit test project.
+                if (testResults.Count() < _testCases.Count())
```

Comment:
For every testrun all testcases are being count. It looks like a simple integer property would do the job too.